File: 1-TwoSum/1-TwoSum.java
// Last updated: 3/30/2026, 2:14:44 PM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int index = 0 ; index < nums.length ; index++){
            int value = target - nums[index];
            if(map.containsKey(value)){
                result[0] = map.get(value);
                result[1] = index;
                break;
            }
            map.put(nums[index],index);
        }
        return result;
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 3/30/2026, 2:14:43 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        unordered_map<int,int> hashmap;
        for(int index = 0 ; index < nums.size() ; index++){
            int value = target - nums[index];
            if(hashmap.find(value)!=hashmap.end()){
                result.push_back(hashmap[value]);
                result.push_back(index);
            }
            hashmap[nums[index]] = index;
        }
        return result;
    }
};
